source/GameInterface/Services/Villages/Handlers/VillageHandler.cs

```diff
@@ -0,0 +1,113 @@
+    public VillageHandler(IMessageBroker messageBroker, IObjectManager objectManager)
+    {
+        this.messageBroker = messageBroker; 
+        this.objectManager = objectManager;
+ 
+        messageBroker.Subscribe<ChangeVillageState>(HandleVillageState);
+        messageBroker.Subscribe<ChangeVillageTradeBound>(HandleTradeBound);
+        messageBroker.Subscribe<ChangeVillageHearth>(HandleHearth);
+        messageBroker.Subscribe<ChangeVillageTradeTaxAccumulated>(HandleTradeTax);
+        messageBroker.Subscribe<ChangeVillageLastDemandTime>(HandleLastDemandTime);
+    }
+
+    private void HandleLastDemandTime(MessagePayload<ChangeVillageLastDemandTime> payload)
+    {
+        var obj = payload.What;
+
+
+        if (objectManager.TryGetObject<Village>(obj.VillageId, out var village) == false)
+        {
+            Logger.Error("Unable to find Village ({villageId})", obj.VillageId);
+            return;
+        }
+
+        VillagePatches.RunLastDemandTimeSatisified(village, obj.LastDemandSatifiedTime);
+    }
+
+    private void HandleTradeTax(MessagePayload<ChangeVillageTradeTaxAccumulated> payload)
+    {
+        var obj = payload.What;
+
+
+        if (objectManager.TryGetObject<Village>(obj.VillageId, out var village) == false)
+        {
+            Logger.Error("Unable to find Village ({villageId})", obj.VillageId);
+            return;
+        }
+
+        VillagePatches.RunTradeTaxChange(village, obj.TradeTaxAccumulated);
+
+    }
+
+    private void HandleHearth(MessagePayload<ChangeVillageHearth> payload)
+    {
+        var obj = payload.What;
+
+        if (objectManager.TryGetObject<Village>(obj.VillageId, out var village) == false)
+        {
+            Logger.Error("Unable to find Village ({villageId})", obj.VillageId);
+            return;
+        }
+
+        VillagePatches.ChangeHearth(village, obj.Hearth);
+    }
+
+    private void HandleTradeBound(MessagePayload<ChangeVillageTradeBound> payload)
+    {
+        var obj = payload.What;
+
+        if (objectManager.TryGetObject<Village>(obj.VillageId, out var village) == false)
+        {
+            Logger.Error("Unable to find Village ({villageId})", obj.VillageId);
+            return;
+        }
+
+        if (objectManager.TryGetObject<Settlement>(obj.TradeBoundID, out var settlement) == false)
+        {
+            Logger.Error("Unable to find Village ({villageId})", obj.VillageId);
+            return;
+        }
+
+
+        VillagePatches.RunTradeBoundChange(village, settlement);
+    }
+
+    private void HandleVillageState(MessagePayload<ChangeVillageState> payload)
+    {
+        var obj = payload.What;
+
+        if(objectManager.TryGetObject<Village>(obj.VillageId, out var village) == false)
+        {
+            Logger.Error("Unable to find Village ({villageId})", obj.VillageId);
+            return;
+        }
+
+        VillagePatches.RunVillageStateChange(village, (Village.VillageStates)obj.State);
+    }
+
+
+    public void Dispose()
```

Comment:
Add all subscriptions to the dispose method, I recommend having Dispose right below the constructor to make this more obvious